apis/cache/v1alpha1/cachebackend_types.go

```diff
@@ -17,6 +17,8 @@ limitations under the License.
 package v1alpha1
```

Comment:
I was thinking the naming between `CacheBackend` and `Dataset`, `CacheBackend` couples with caching system semantics but `Dataset` only describes a set of data with higher-level abstraction, cache can be an add-on feature over dataset.

Reply:
Hi Simon, you are right. Maybe we can merge this PR first? I will refactor CacheBackend to Dataset in another PR.